homeassistant/components/speedtestdotnet/sensor.py

```diff
@@ -139,5 +135,5 @@ def extra_state_attributes(self) -> dict[str, Any]:
     async def async_added_to_hass(self) -> None:
         """Handle entity which will be added."""
         await super().async_added_to_hass()
-        if state := await self.async_get_last_state():
-            self._state = state.state
+        if state := await self.async_get_last_sensor_data():
+            self._state = cast(StateType, state.native_value)
```

Comment:
Please revert the `StateType` change.

Reply:
Will do, but then there's still an issue that I can't make this testable since at start of the integration, a speedtest is being executed, which makes testing this not possible I think?

Reply:
Mock it out?

Reply:
Still think we should change this PR to remove RestoreEntity as I don't understand why we would have it when using a DUC eg we have a new value once init so why restore an old value.